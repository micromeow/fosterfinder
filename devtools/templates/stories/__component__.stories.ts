import type { Meta, StoryObj } from '@storybook/react';
// @ts-ignore
import { __component__ } from '@/components/ui/__component__';

const meta: Meta<typeof __component__> = {
  title: 'Example/__component__',
  component: __component__,
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof __component__>;

export const Primary: Story = {
  args: {},
};
