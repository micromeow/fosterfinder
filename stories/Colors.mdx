{/* Colors.mdx */}

import { Meta, ColorPalette, ColorItem } from "@storybook/blocks";

<Meta label="Colors" />

<h1>Colors</h1>

<ColorPalette>
  <ColorItem
    title="theme.color.greyscale"
    subtitle="Some of the greys"
    colors={{ White: "#FFFFFF", Alabaster: "#F8F8F8", Concrete: "#F3F3F3" }}
  />
  <ColorItem
    title="theme.color.positive"
    subtitle="Green"
    colors={{
      Apple: "rgba(102,191,60,1)",
      Apple80: "rgba(102,191,60,.8)",
      Apple60: "rgba(102,191,60,.6)",
      Apple30: "rgba(102,191,60,.3)",
    }}
  />
</ColorPalette>
