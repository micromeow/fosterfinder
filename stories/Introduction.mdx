{/* Introduction.mdx */}

import { Meta, Source } from "@storybook/blocks";

<Meta label="Introduction" />

<h1>Introduction</h1>
<h2>READ ME BEFORE DOING ANYTHING</h2>
<p>
  This is the introduction to the component library for Foster Finder. It is built on
  the <a href="https://ui.shadcn.com/">shadcn/ui</a> component library, an open
  sourced component library built on{" "}
  <a href="https://tailwindcss.com">tailwindcss</a>.
</p>

<h2>Installing New Components from `shadcn`</h2>
<p>
  shadcn uses a "need-to-have" approach to adding new components. If you need to
  use a component that has not been installed in the `components/ui` directory,
  follow these steps to install it with the correct configuration.
</p>

1 - Install the component using the following command:

<Source code={`npx shadcn-ui@latest add <component-name>`} />

2 - <b>IMPORTANT</b>: Rename the generated component to use PascalCase. For example, if you are adding the component `button`, rename the component to `Button`.

3 - Generate stories and tests for the component using the following command:

<Source code={`npm run codegen`} />
