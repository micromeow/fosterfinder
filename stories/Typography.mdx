{/* Typography.mdx */}

import { Meta } from "@storybook/blocks";

<Meta label="Typography" />

<h1>Typography</h1>

<a href="https://ui.shadcn.com/docs/components/typography">
  https://ui.shadcn.com/docs/components/typography
</a>
