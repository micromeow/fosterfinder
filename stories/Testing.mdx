{/* Introduction.mdx */}

import { Meta, Source } from "@storybook/blocks";

<Meta label="Testing" />

<h1>Testing</h1>

Each component should at minimum have a snapshot test. This ensures that the
component is rendered correctly and that it doesn't change unexpectedly. To
run the tests, run the following command:

<Source code={`npm test`} />

<h2>Creating a test</h2>

You may use the following command to generate a test and storybook file for a component:

<Source code={`npm run codegen`} />

Or you may do it manually:

<Source code={`
import * as stories from "./Button.stories";
import { render } from "@testing-library/react";
import { composeStories } from "@storybook/react";

const asyncButtonStories = composeStories(stories);

const testCases = Object.values(asyncButtonStories).map((Story) => [
	// The ! is necessary in Typescript only, as the property is part of a partial type
	Story.storyName!,
	Story,
]);

test.each(testCases)("Renders %s story", async (\_storyName, Story) => {
	const tree = await render(<Story />);
	expect(tree.baseElement).toMatchSnapshot();
});
`}/>
