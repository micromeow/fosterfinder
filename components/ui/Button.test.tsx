import * as stories from './Button.stories';
import { render } from '@testing-library/react';
import { composeStories } from '@storybook/react';

const asyncButtonStories = composeStories(stories);

const testCases = Object.values(asyncButtonStories).map((Story) => [
  // The ! is necessary in Typescript only, as the property is part of a partial type
  Story.storyName!,
  Story,
]);

test.each(testCases)('Renders %s story', async (_storyName, Story) => {
  const tree = await render(<Story />);
  expect(tree.baseElement).toMatchSnapshot();
});
