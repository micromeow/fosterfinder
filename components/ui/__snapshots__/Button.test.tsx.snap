// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Primary story 1`] = `
<body>
  <div>
    <button
      class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:-translate-y-0.5 transition bg-primary text-primary-foreground h-10 px-4 py-2"
    >
      Button
    </button>
  </div>
</body>
`;
